code/modules/mob/living/carbon/life.dm

```diff
@@ -217,6 +218,35 @@
 	if (breath_gases[/datum/gas/nitryl])
 		var/nitryl_partialpressure = (breath_gases[/datum/gas/nitryl][MOLES]/breath.total_moles())*breath_pressure
 		adjustFireLoss(nitryl_partialpressure/4)
+	//MIASMA
+	if (breath_gases[/datum/gas/miasma])
+		var/miasma_partialpressure = (breath_gases[/datum/gas/miasma][MOLES]/breath.total_moles())*breath_pressure
+
+		if(prob(1 * miasma_partialpressure))
+			var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+			miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)
+			miasma_disease.try_infect(src)
+
+		switch(miasma_partialpressure)
+			if(0.1 to 0.5)
+				// At lower pp, give out a warning
+				if(prob(5))
+					to_chat(src, "<span class='notice'>There is an unpleasant smell in the air.</span>")

```

Comment:
I hate myself for mentioning this, but these should probably have moodlet effects

Reply:
I'll have a look at it later

Reply:
Did it, but I'm unsure whether I did it right. Can you please have a look at it once the checks have all been done?